internal/planner/planner.go

```diff
@@ -24,12 +24,17 @@ type wasmBuiltin struct {
 
 // internalBuiltins are the built-in functions implemented in wasm.
 var internalBuiltins = map[string]wasmBuiltin{
-	ast.Plus.Name:     wasmBuiltin{ast.Plus, "opa_arithmetic_plus"},
-	ast.Minus.Name:    wasmBuiltin{ast.Minus, "opa_arithmetic_minus"},
-	ast.Multiply.Name: wasmBuiltin{ast.Multiply, "opa_arithmetic_multiply"},
-	ast.Divide.Name:   wasmBuiltin{ast.Divide, "opa_arithmetic_divide"},
-	ast.Abs.Name:      wasmBuiltin{ast.Abs, "opa_arithmetic_abs"},
-	ast.Rem.Name:      wasmBuiltin{ast.Rem, "opa_arithmetic_rem"},
+	ast.Plus.Name:         wasmBuiltin{ast.Plus, "opa_arithmetic_plus"},
+	ast.Minus.Name:        wasmBuiltin{ast.Minus, "opa_arithmetic_minus"},
+	ast.Multiply.Name:     wasmBuiltin{ast.Multiply, "opa_arithmetic_multiply"},
+	ast.Divide.Name:       wasmBuiltin{ast.Divide, "opa_arithmetic_divide"},
+	ast.Abs.Name:          wasmBuiltin{ast.Abs, "opa_arithmetic_abs"},
+	ast.Rem.Name:          wasmBuiltin{ast.Rem, "opa_arithmetic_rem"},
+	ast.ArrayConcat.Name:  wasmBuiltin{ast.ArrayConcat, "opa_array_concat"},
+	ast.ArraySlice.Name:   wasmBuiltin{ast.ArraySlice, "opa_array_slice"},
+	ast.SetDiff.Name:      wasmBuiltin{ast.SetDiff, "opa_set_diff"},
+	ast.Intersection.Name: wasmBuiltin{ast.Intersection, "opa_sets_intersection"},
```

Comment:
Do we need to register one for `ast.Or` and `ast.And` for the `opa_set_union` and `opa_set_intersection`? Or are those already being handled elsewhere (like the ast.Minus for the diff).

Reply:
I missed their registration in topdown/binary.go. Fixed.